### DISCUSSION TITLE
Get pseudo-bulk coverage files

### DISCUSSION THREAD

**Discussion by Dazcam** [GreenleafLab/ArchR#969] [Questions / Documentation]
Is there a method to view/manipulate the pseudo-bulk coverage files generated by ArchR within Rstudio? I see that the relevant h5 fragment files are in the Group coverages directory in my ArchR project but I've had a look through the list of `Get` functions on your reference page and I don't see anything relevant (though I may have missed the function).

Is there an ArchR function/method I could use to access these files within my Rstudio session?

Many Thanks.

**Comment by rcorces** [comment]
Can you clarify what you mean by pseudo-peak files? Do you mean the pseudo-bulk coverage files or the peak files that result from calling peaks on those pseudo-bulk coverage files? Or something else perhaps?

**Reply by Dazcam** [reply]
Apologies, I'm referring to the pseudo-bulk coverage files. I've amended the OP.

**Reply by rcorces** [reply]
Unfortunately, we dont provide any functions to manipulate those coverage files. I think that is the safest implementation to prevent end users from messing with a file that is important for downstream analysis.
Have you looked at the `getGroupBW()` function? This exports a bigWig file that is similar (though not the same replicates etc) as the coverage files and you could edit that.
https://www.archrproject.com/reference/getGroupBW.html

If you have a particular application in mind, feel free to let us know and we will assess whether it makes sense to implement.

**Reply by Dazcam** [reply]
No problem. Thank you for responding. I suspected that might be the case. I was looking to export the files (in some way) to run on a different peak caller. I can see you have a function for converting the coverage files to bed files in preparation for peak calling in `macs2`. Are these coverage bed files stored anywhere?

**Reply by rcorces** [reply]
The bed files are generated and then removed automatically in the `.callSummitsOnCoverages()` hidden function via the `.writeCoverageToBed()` hidden function.

You should be able to use this function via `ArchR:::.writeCoverageToBed()` to generate the bed files you want
https://github.com/GreenleafLab/ArchR/blob/968e4421ce7187a8ac7ea1cf6077412126876d5f/R/GroupCoverages.R#L739
